frontend/model/contracts/group.js

```diff
@@ -339,6 +360,52 @@ DefineContract({
         archiveProposal(state, data.proposalHash)
       }
     },
+    'gi.contracts/group/removeMember': {
+      validate: (data, { state, getters }) => {
+        objectOf({
+          member: string,
+          memberId: string,
+          groupId: string,
+          // In case it happens in a big group (by proposal)
+          // we need to validate the associated proposal.
+          proposalHash: optional(string),
+          proposalPayload: optional(objectOf({
+            secret: string // NOTE: simulate the OP_KEY_* stuff for now
+          }))
+        })(data)
+
+        const memberToRemove = data.member
+        const membersCount = getters.groupMembersCount
+
+        if (!state.profiles[memberToRemove]) {
+          throw new TypeError('Not part of the group.')
+        }
+        if (membersCount === 1) {
+          throw new TypeError('Cannot remove yourself.')
+        }
+        if (membersCount >= 3) {
+          const { payload } = state.proposals[data.proposalHash] || {}
+          if (!payload || payload.secret !== data.proposalPayload.secret) {
+            throw new TypeError('Invalid associated proposal.')
+          }
+        }
+      },
+      process ({ data, meta }, { state }) {
+        const rootState = sbp('state/vuex/state')
+        if (data.member === rootState.loggedIn.username) {
+          // If this member is re-joining the group, ignore the rest
+          // so the member doesn't remove themself again.
+          if (sbp('okTurtles.data/get', 'JOINING_GROUP')) {
+            return false
+          }
+        }
+
+        Vue.delete(state.profiles, data.member)
+      },
+      sideEffect ({ data }) {
+        removeMemberSideEffect(data)
```

Comment:
1. `sideEffect` must be marked `async`
2. `removeMemberSideEffect` must have an `await` before it (here and anywhere else it's called, e.g. proposals.js)